### PR TITLE
[RFR] Add invite to create the first record when the list is empty

### LINKED ISSUE #2639: Empty List Invite
If there is no result, and if there is no active filter, and if the resource has a Create page, the List page should display a special content inviting the user to create their first element.

localhost_8080_

Inspirations:

- http://emptystat.es/
- https://uxplanet.org/empty-state-mobile-app-nice-to-have-essential-f11c29f01f3
- https://twitter.com/steveschoger/status/1058398981888376832

### PR DESCRIPTION
**Pull request by m4theushw**
Closes #2639 

- [X] Invite to create the first record
- [x] Add tests
- [x] French translations
- [x] Write docs showing how to override `<EmptyState />`

empty-state

### CODE REVIEW

**Comment by fzaninotto** [comment]
Very encouraging!

**Comment by fzaninotto** [comment]
Could you PR against `next` rather than `master`?

File: docs/List.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Here are all the props accepted by the `<List>` component:
 * [`filterDefaultValues`](#filter-default-values) (the default values for `alwaysOn` filters)
 * [`pagination`](#pagination)
 * [`aside`](#aside-component)
+* [`emptyState`](#empty-state)
```
Comment (fzaninotto) [inline docs/List.md]: I'd call it `empty` instead

File: docs/List.md [hunk 2]
```diff
@@ -683,6 +685,34 @@ const Aside = ({ data, ids }) => (
 ```
 {% endraw %}
 
+### Empty state
+
+When there is no result, and there is no active filter, and the resource has a create page then a special page inviting the user to create the first record is displayed. Use the `emptyState` prop to modify that page, passing your custom component:
```
Comment (fzaninotto) [inline docs/List.md]:
```suggestion
When there is no result, and there is no active filter, and the resource has a create page, react-admin displays a special page inviting the user to create the first record.

You can use the `emptyState` prop to replace that page by a custom component:
```

File: docs/List.md [hunk 3]
```diff
@@ -683,6 +685,34 @@ const Aside = ({ data, ids }) => (
...
+
+const ProductList = props => (
+    <List emptyState={<EmptyState />} {...props}>
+        ...
+    </List>
+);
+```
+{% endraw %}
+
+The `emptyState` component receives the same props as the `aside` prop. Read the [section above](#aside-component) to check them.
```
Comment (fzaninotto) [inline docs/List.md]: no, please repeat them - developers don't read the docs sequentially.

File: docs/List.md [hunk 4]
```diff
@@ -683,6 +685,34 @@ const Aside = ({ data, ids }) => (
...
+When there is no result, and there is no active filter, and the resource has a create page then a special page inviting the user to create the first record is displayed. Use the `emptyState` prop to modify that page, passing your custom component:
+
+{% raw %}
+```jsx
+import Button from '@material-ui/core/Button';
+import { CreateButton, List } from 'react-admin';
+
+const EmptyState = ({ basePath }) => (
+    <div style={{ textAlign: 'center', margin: '1em' }}>
+        <h1>No products available</h1>
```
Comment (fzaninotto) [inline docs/List.md]: I'd use `resource` in the example

File: packages/ra-ui-materialui/src/list/EmptyState.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+});
+
+const EmptyState = ({ resource, basePath }) => {
+    const classes = useStyles();
+    const translate = useTranslate();
+
+    const resourceName = translate(`resources.${resource}.name`, {
+        smart_count: 0,
+        _: inflection.humanize(resource),
+    }).toLowerCase();
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/EmptyState.js]: We don't use `toLowerCase` anywhere else on the naming, and I `humanize` should do the trick.

File: packages/ra-ui-materialui/src/list/List.js [hunk 6]
```diff
@@ -117,20 +119,24 @@ export const ListView = props => {
...
     const controllerProps = getListControllerProps(rest);
 
-    return (
-        <div
-            className={classnames('list-page', classes.root, className)}
-            {...sanitizeRestProps(rest)}
-        >
-            <Title title={title} defaultTitle={defaultTitle} />
-
+    const renderChildren = () => (
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/List.js]: renderList?

**Comment by fzaninotto** [comment]
Why did you close your PR?

**Comment by m4theushw** [comment]
@fzaninotto I will open another PR targeting `next`. To use this one I would have to first  merge with `master` since the branches are out-of-sync.

**Pull request closed without merging**